emcc.py

```diff
@@ -2330,6 +2330,14 @@ def phase_linker_setup(options, state, newargs):
       # included, as the entire JS library can refer to things that require
       # these exports.)
       settings.REQUIRED_EXPORTS += [
+        'wasmfs_create_memory_backend',
+        'wasmfs_create_node_backend',
+        'wasmfs_create_opfs_backend',
+        'wasmfs_create_js_file_backend',
+        'wasmfs_create_fetch_backend',
+        'wasmfs_create_icase_backend_from_pointer',
```

Comment:
Should putting something in the __deps be expected to automatically import it?

I've tried a few different variations, such as __deps: ['wasmfs_create_node_backend'], __deps: ['_wasmfs_create_node_backend'], and __deps: ['$wasmfs_create_node_backend'], but none of them correctly imported the function unless I also added wasmfs_create_node_backendtoREQUIRED_EXPORTSinemcc.py`.

As a side note, it looks like the library_wasmfs_*.js files for the other backends are included on every execution. I added the globals to those files, but I'm wondering if it might be better to directly create library_opfs.js etc. to use with options like -lopfs.js and move the globals there. Basically, would that work as expected?

Reply:
I think that adding native functions to  `__deps` only currently works when the JS function is called/needed by that native code.

If you JS function in question is only called from JS, or included via `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` then this mechanism won't work.   We should really fix that.

Reply:
I see. Is there a way for me to conditionally add the function to exports in emcc.py if -lnodefs.js is set on the command line?

Reply:
I found the bug we have open for this: https://github.com/emscripten-core/emscripten/issues/19759

Reply:
@jameshu15869 

> Is there a way for me to conditionally add the function to exports in emcc.py if -lnodefs.js is set on the command line?

Technically that information is accessible:

```diff
diff --git a/emcc.py b/emcc.py
index 5e347e303..d577ebda4 100755
--- a/emcc.py
+++ b/emcc.py
@@ -4238,6 +4240,7 @@ def process_libraries(state, linker_inputs, embind_emit_tsd):
     new_flags.append((i, flag))
 
   settings.JS_LIBRARIES += libraries
+  print('debug', settings.JS_LIBRARIES)
 
   # At this point processing JS_LIBRARIES is finished, no more items will be added to it.
   # Sort the input list from (order, lib_name) pairs to a flat array in the right order.
```
But that is at the link stage, which is after the compile flags stuff that has the main WasmFS logic that sets up the required exports. So putting it here would be a little odd.

@sbc100 do you have a better idea for a workaround?

Reply:
Not really I'm afraid.  I can work on #19759 so maybe we can just take the code size hit until that fix lands.

BTW, You even you put the dependencies here because they don't work in the `__deps` .. we can/should still add them to the `__deps` since its a good failsafe and it expresses the dependency in the correct place.